src/js/serial.js

```diff
@@ -112,13 +108,19 @@ var serial = {
                                 });
                             }, 50);
                             break;
-                            
+
                         case 'timeout':
-                            // TODO
+                            // No data has been received for receiveTimeout milliseconds.
+                            // We will do nothing.
                             break;
-                            
+
                         case 'break': // This seems to be the error that is thrown under NW.js in Windows when the device reboots after typing 'exit' in CLI
+                        case 'disconnected':
```

Comment:
Not entirely sure we want to treat these all the same - `disconnected` and `device_lost` seem to be 'normal' errors that are to be expected when disconnecting a board.
Maybe we should log `frame_error` and `parity_error` as errors?

Reply:
console.log is usually hidden from the user. do you want to GUI.log `frame_error` and `parity_error`? I've gotten some `frame_error` but haven't gotten `parity_error` yet.

Reply:
I'd probably `GUI.log` these, to make the user aware that there is a problem (going from issues that are opened, a bad cable in most cases).